src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -589,6 +589,20 @@
     <Field Name="ConstantValueOpt" Type="ConstantValue?"/>
   </Node>
 
+  <!--
+  Default expressions can convert to the target type.
+  Once converted to a target type, they cannot be target-typed again.
+  The Converted default expression is one which has been already converted to a target type.
+  Converted default expressions always has a type.
+  Does not survive initial binding.
+  -->
+  <Node Name="BoundConvertedDefaultExpression" Base="BoundDefaultExpression">
+    <!-- Converted default expression must have a type. -->
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <!-- Natural type is preserved for the purpose of semantic model. Can be `null`. -->
+    <Field Name="NaturalTypeOpt" Type="TypeSymbol" Null="allow"/>
```

Comment:
><Field Name="NaturalTypeOpt" Type="TypeSymbol" Null="allow"/> [](start = 4, length = 61)

Is this property useful? Is it ever not null?

Reply:
`default(T)` has a natural type.

---
In reply to: [314033937](https://github.com/dotnet/roslyn/pull/37596#discussion_r314033937) [](ancestors = 314033937)

Reply:
> default(T) has a natural type.

But it doesn't have any other type.

---
In reply to: [314093546](https://github.com/dotnet/roslyn/pull/37596#discussion_r314093546) [](ancestors = 314093546,314033937)